### PR TITLE
Fixed widgets using echo in their run. #7417

### PR DESCRIPTION
**Pull request by SamMousa**
Fixed all widgets that do not generate output in both their init() and run() classes.

### CODE REVIEW

**Comment by samdark** [comment]
Collapse widget is broken. See test.

**Comment by samdark** [comment]
I guess that's because of trailing `\n` is missing now.

**Comment by samdark** [comment]
Same about other occurances.

**Comment by SamMousa** [comment]
The trailing \n wasn't there in every widget, I've readded it to pass the tests.

**Comment by samdark** [comment]
Thanks!

File: extensions/bootstrap/ButtonDropdown.php [hunk 1]
```diff
@@ -79,11 +79,13 @@ public function run()
         $options = $this->containerOptions;
         $tag = ArrayHelper::remove($options, 'tag', 'div');
 
-        echo Html::beginTag($tag, $this->containerOptions);
-        echo "\n" . $this->renderButton();
-        echo "\n" . $this->renderDropdown();
-        echo "\n" . Html::endTag($tag);
         $this->registerPlugin('button');
+        return implode("\n", [
```
Comment (Faryshta) [inline extensions/bootstrap/ButtonDropdown.php]: honest question. Why is this required? the code generated here is not to be debuged or read by humans since is dinamically generated.

Comment (samdark) [inline extensions/bootstrap/ButtonDropdown.php]: You mean `"\n"`? Just in case someone will actually read it, I guess. No real reason.

Comment (Faryshta) [inline extensions/bootstrap/ButtonDropdown.php]: wouldn't it be faster and less memory demanding to remove it completely?

specially since the view, layouts and some widgets are captured in a `ob_start()`

Comment (samdark) [inline extensions/bootstrap/ButtonDropdown.php]: Not really. It doesn't take much time/memory to do `implode`.

Comment (Faryshta) [inline extensions/bootstrap/ButtonDropdown.php]: how how about a configurable separator?

Comment (samdark) [inline extensions/bootstrap/ButtonDropdown.php]: It's not something that should be configurable.

Comment (Faryshta) [inline extensions/bootstrap/ButtonDropdown.php]: but it takes memory in the sense that `echo` is captured by `ob_start` repeatedly

Comment (samdark) [inline extensions/bootstrap/ButtonDropdown.php]: Currently not.

Comment (SamMousa) [inline extensions/bootstrap/ButtonDropdown.php]: Ob_start makes echo faster not slower. That's why it's enabled by default
in recent php versions.

Nonetheless line endings are unneeded but readable HTML doesn't hurt.

Same for implode vs string concatenation, readable php is worth something
IMO.
